File: Medium/Java/45. Jump Game II.java
class Solution {
    public int jump(int[] nums) {
        // Initial values
        int max = 0;
        int jump = 0;
        int count = 0;
        
        // Use a for loop to check how far we can jump
        for(int i = 0; i < nums.length - 1; i++){
            max = Math.max(max, i + nums[i]);
            if(jump == i){
                jump = max;
                count++;
            }
        }

        // Return the number of jumps taken
        return count;
    }
}

File: Medium/Python/45. Jump Game II.py
class Solution(object):
    def jump(self, nums):
        # Initial values
        maximum = 0
        jump = 0
        count = 0

        # Use a foor loop to check how far we can jump
        for i in range(len(nums) - 1):
            maximum = max(maximum, i + nums[i])
            if jump == i:
                jump = maximum
                count += 1
        
        # Return the number of jumps taken
        return count
